org.eclipse.scanning.api/src/org/eclipse/scanning/api/device/models/DeviceWatchdogModel.java

```diff
@@ -1,219 +1,219 @@
+	{@literal 	<property name="topupTime"              value="15000"/>}
+	{@literal   <!-- End optional, do not usually need to set -->}
+
+    {@literal   <property name="bundle"               value="org.eclipse.scanning.api" /> <!-- Delete for real spring? -->}
+	{@literal </bean>}
+	{@literal <bean id="topupWatchdog"    class="org.eclipse.scanning.sequencer.watchdog.TopupWatchdog" init-method="activate">}
+	{@literal 	<property name="model"    ref="topupModel"/>}
+    {@literal   <property name="bundle"   value="org.eclipse.scanning.sequencer" /> <!-- Delete for real spring? -->}
+	{@literal </bean>}
+    </pre>
+
+
+<h3>Calculation of scannable parts of topup  </h3>
+    <pre>
+
+    |<-w->|
+    |.
+    |  .
+    |    .
+    |      .
+    |        .
+    |          .
+    |            .
+    |              .|<-   c  ->|
+    |                .
+    |                  .
+    |                    .
+    |                      .
+    |                        . |<-Tf->|
+    |                          ........    t
+    |
+    |__________________________________(time)
+
+    |<-              p              ->|
+
+    w  - warmup
+    c  - cooloff
+    t  - time until next topup next occurs
+    Tf - Topup fill time (variable but max 15s in normal mode)
+    p  - Period of cycle, usually 10mins or so.
+
+    In order for scanning to run, all of the following conditions must be satisfied:
+
+    1. Mode is normal (8)
+    2. t > c
+    3. t < (p-Tf)-w
+
+    </pre>
+
+<h3>Ring Mode</h3>
+
+The "Ring Mode" PV is SR-CS-RING-01:MODE.
+
+This PV has various states:
+<img src="./doc/modes.png" />
+
+In brief though, the only one you need to care about is state 8 = VMX. This is "normal" mode now that we've installed the new VMX (AKA DDBA) components.
+
+If this PV = 8, then we're in normal mode. If this PV is anything else, then we're in some other state.
+
+ * @author Matthew Gerring
+ *
+ */
+public class DeviceWatchdogModel {
+
+	/**
+	 * This is an expression of scannables which are monitored and the expression
+	 * reevaluated if they change during a scan. Each variable in the scan *must*
+	 * be a scannable name. The expression will be evaluated using JEXL which is
+	 * available through the IExpressionService. Any expression
+	 */
+	private String expression; // e.g. 'beamcurrent >= 1.0 && !portshutter.equalsIgnoreCase("Closed")'
+	private String message;
+
+	// t
+	private String countdownName; // e.g. "topup", "countdown" PV likely to be SR-CS-FILL-01:COUNTDOWN which is in s
+
+	// c in ms
+	private long   cooloff;       // time in ms before topup for which the scan should be paused.
+
+	// w in ms
+	private long   warmup;        // time in ms after topup the scan should wait before starting.
+
+	// p in ms
+	private long   period = 10*60*1000; // Period in ms, default is 10min
+
+	// Tf in ms
+	private long   topupTime = 15*1000; // The time that a topup takes. This is varible but in normal mode <= 15s
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Cast one of the operands of this multiplication operation to a "long". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2184)